enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
```

Comment:
1. enumerables #my_each_with_index returns an enumerator if no block is given
2. enumerables #my_each_with_index when a block is given and self is an array it should return the array itself after calling the given block once for each element in self
3. enumerables #my_each_with_index when a block is given when self is a range, it should return the range itself after calling the given block once for each element in self

Reply:
Solved all the requested changes